lib/trino-parquet/src/main/java/io/trino/parquet/reader/ParquetReader.java

```diff
@@ -224,6 +250,30 @@ public void close()
         freeCurrentRowGroupBuffers();
         currentRowGroupMemoryContext.close();
         dataSource.close();
+
+        if (writeChecksumBuilder.isPresent()) {
+            ParquetWriteValidation parquetWriteValidation = writeValidation.orElseThrow();
+            parquetWriteValidation.validateChecksum(dataSource.getId(), writeChecksumBuilder.get().build());
+        }
+    }
+
+    public Page nextPage()
+            throws IOException
+    {
+        int batchSize = nextBatch();
+        if (batchSize <= 0) {
+            return null;
```

Comment:
under what circumstances is this possible?

btw maybe we should throw here?
returning `null` will cause `ParquetPageSource` to close and not read anything anymore:
https://github.com/trinodb/trino/blob/f8e774a949773399df5fa823186e4a68d79b931d/plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetPageSource.java#L103-L105

cc @alexjo2144 @homar

Reply:
This is just preserving the old code's behaviour
```
int batchSize = parquetReader.nextBatch();

if (closed || batchSize <= 0) {
      close();
      return null;
}
```
batchSize can be -1 on reaching end of list of row groups to read, which is okay.
batchSize can be 0 on encountering empty row groups. This should be encountered only for an empty file. It's possible that some buggy writer writes empty row groups in the middle of a file with data, in that case we will stop early.